### PR TITLE
Ensures extension can run scripts

### LINKED ISSUE #281: Print option is not working in report page
Adding screenshot for reference. Not sure if this print option is in progress or not

screen shot 2016-05-05 at 9 17 56 am

### PR DESCRIPTION
**Pull request by paullewis**
* Copies scripts to extension/pages folder
* Sets extension HTML report to use external script
* Adds `inline` flag for the CLI

Fixes #281

### CODE REVIEW

File: extension/app/pages/scripts/report.js [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+  this.addEventListeners();
+  this.onResize();
+}
+
+Report.prototype = {
+  onScroll: function() {
+    this.handleMenuStickiness();
+  },
+
+  handleMenuStickiness(options) {
```
Comment (samccone) [inline extension/app/pages/scripts/report.js]: nit: this is chrome so we might as well do `options={}`

File: extension/app/pages/scripts/report.js [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    var menuShouldBeSticky = (window.scrollY > this.menuTopPosition);
+
+    // If no change early exit.
+    if (!forceUpdate && menuShouldBeSticky === menuIsSticky) {
+      return;
+    }
+
+    if (menuShouldBeSticky) {
+      this.menu.classList.add('menu--fixed');
+      this.menu.style.width = this.menuBCR.width + 'px';
```
Comment (samccone) [inline extension/app/pages/scripts/report.js]: nit: string interpolation

**Comment by paullewis** [comment]
I opted for older ES5 support here, since I imagine it's likely that someone will send the report to a colleague or someone else who may not be using Chrome, but is (hopefully) using a fairly recent browser. I intentionally opted for less modern JS so as not to lock out those users.

**Comment by paulirish** [comment]
There are 3 unique files named report.js? Little confused here, and if one of them is a copy of the stuff at in the root `/report/`.

File: extension/gulpfile.js [hunk 3]
```diff
@@ -30,6 +31,15 @@ gulp.task('extras', () => {
   .pipe(gulp.dest('dist'));
 });
 
+gulp.task('copyReportScripts', () => {
+  return gulp.src([
+    '../report/scripts/report.js'
+  ])
+  .pipe(rename('pages/scripts/report.js'))
```
Comment (paulirish) [inline extension/gulpfile.js]: If this is a copy, should `pages/scripts/report.js` be in .gitignore?

Comment (paullewis) [inline extension/gulpfile.js]: Yeah it can be. I'll add it.

**Comment by paullewis** [comment]
> There are 3 unique files named report.js? Little confused here, and if one of them is a copy of the stuff at in the root /report/.

Yeah I'll rename them to be clearer.

**Comment by paullewis** [comment]
Okay:

* the main report generator is now report-generator.js
* the report loader for the extension is now in report-loader.js
* the baked out JS is called lighthouse-report.js

Internally all names changed to match.

**Pull request merged**